src/core/textures/Texture.js

```diff
@@ -216,15 +224,17 @@ export default class Texture extends EventEmitter
     {
         if (this.baseTexture)
         {
-            if (destroyBase)
+            // remove BaseTexture stored under the texture name if destroyBase is true or
+            // is our base texture (in which case this is the event callback for 'destroy')
+            if ((destroyBase === true || destroyBase === this.baseTexture)
```

Comment:
I totally get that it isn't often you reuse the texture. I just don't like that some code for removing BaseTexture from the cache is in Texture and some code for removing Texture from the cache is in BaseTexture.

These two classes are supposed to be separate things, but this code ties them together so much that they almost shouldn't be separate.